app/models/concerns/foreman_openscap/openscap_proxy_core_extensions.rb

```diff
@@ -4,18 +4,22 @@ module OpenscapProxyCoreExtensions
 
     included do
       validate :openscap_proxy_has_feature
+      validate :scap_client_class_present
+      after_save :update_scap_client
     end
 
-    def update_scap_client_params(proxy_id)
-      new_proxy = SmartProxy.find proxy_id
+    def update_scap_client
+      update_scap_client_params if openscap_proxy_id_changed? && openscap_proxy_id
```

Comment:
does this mean that it does not update values if I disassociate proxy? shouldn't it delete lookup values?